nni/contrib/compression/quantization/lsq_quantizer.py

```diff
@@ -116,7 +116,16 @@ def register_scale(self):
             for target_name, _ in ts.items():
                 if hasattr(wrapper, f"{target_name}_scale"):
                     delattr(wrapper, f"{target_name}_scale")
-                param = torch.nn.Parameter()
+                # for deepspeed
+                try:
+                    device = next(wrapper.parameters()).device
+                except StopIteration:
+                    try:
+                        device = next(wrapper.buffers()).device
+                    except StopIteration:
+                        # NOTE: this will have risk in model parallel
+                        device = next(self.bound_model.parameters()).device
+                param = torch.nn.Parameter(torch.Tensor([0.01]).to(device))
```

Comment:
does other quantizers also need such changes? or only lsq supports deepspeed evaluator?

Reply:
Yes, only the LSQ and LSQ+ quantization algorithms require this because only these two algorithms add extra parameters. DeepSpeed needs to allocate buffers for parameters in advance. If you don't change it according to the above method, you may encounter buffer allocation errors.

Reply:
all quantization and pruning algos support deepspeed evaluator